packages/components/src/Modal/Modal.rebuilt.test.tsx

```diff
@@ -76,41 +76,153 @@ describe("Composable Modal", () => {
   });
-  it('modal contains aria role of "dialog"', async () => {
+
+  it("should render the modal header with custom content", () => {
     render(
-      <Modal.Provider open={true}>
+      <Modal.Provider open={true} modalLabelledBy="custom-header">
         <Modal.Content>
-          <Modal.Header title="Modal Title" />
+          <Modal.Header>
+            <span id="custom-header">Custom Header Content</span>
+          </Modal.Header>
         </Modal.Content>
       </Modal.Provider>,
     );
-    expect(await screen.findByRole("dialog")).toBeInTheDocument();
+    expect(screen.getByRole("dialog")).toHaveAttribute(
+      "aria-labelledby",
+      "custom-header",
+    );
+    expect(screen.getByRole("dialog")).toHaveTextContent(
+      "Custom Header Content",
+    );
   });
 
-  it("names the dialog from title via aria-labelledby when ariaLabel is not provided", async () => {
-    render(
-      <Modal.Provider open={true}>
-        <Modal.Content>
-          <Modal.Header title="Billing Settings" />
-        </Modal.Content>
-      </Modal.Provider>,
-    );
+  describe("accessibility", () => {
+    it('modal contains aria role of "dialog"', async () => {
+      render(
+        <Modal.Provider open={true}>
+          <Modal.Content>
+            <Modal.Header title="Modal Title" />
+          </Modal.Content>
+        </Modal.Provider>,
+      );
+      expect(await screen.findByRole("dialog")).toBeInTheDocument();
+    });
 
-    expect(
-      await screen.findByRole("dialog", { name: "Billing Settings" }),
-    ).toBeInTheDocument();
-  });
+    describe("dialog is named from title prop", () => {
+      it("when ariaLabel is not provided", async () => {
+        render(
+          <Modal.Provider open={true}>
+            <Modal.Content>
+              <Modal.Header title="Billing Settings" />
+            </Modal.Content>
+          </Modal.Provider>,
+        );
 
-  it("uses ariaLabel via aria-label only when no title/header is provided", async () => {
-    render(
-      <Modal.Provider open={true} ariaLabel="Add Customer">
-        <Modal.Content>{/* no Modal.Header on purpose */}</Modal.Content>
-      </Modal.Provider>,
-    );
+        const dialog = await screen.findByRole("dialog");
+
+        expect(
+          screen.getByRole("dialog", { name: "Billing Settings" }),
+        ).toBeInTheDocument();
+        expect(dialog).toHaveAttribute("aria-labelledby");
+      });
+
+      it("when ariaLabel is provided (title takes precedence)", async () => {
+        render(
+          <Modal.Provider open={true} ariaLabel="This should be ignored">
+            <Modal.Content>
+              <Modal.Header title="Billing Settings" />
+            </Modal.Content>
+          </Modal.Provider>,
+        );
+
+        const dialog = await screen.findByRole("dialog");
+
+        expect(dialog).toBeInTheDocument();
+        expect(
+          screen.getByRole("dialog", { name: "Billing Settings" }),
+        ).toBeInTheDocument();
```

Comment:
finding the dialog twice in 2 different ways doesn't really tell us anything. let's just query for it with the method that is most relevant to this test, that it has the precise name we are looking for